eng/pipelines/templates/steps/setup-msvc.yml

```diff
@@ -0,0 +1,17 @@
+
+steps:
+  - pwsh: |
+      $tempFile = New-TemporaryFile;
+      systeminfo.exe /fo csv > $tempFile
+      $osName = (Import-Csv $tempFile).'OS Name'
+
+      Write-Host "OS Name: $osName"
+      if (!($osName -like '*2022*')) {
+        Write-Host "OS is not Windows Server 2022, skipping MSVC changes"
```

Comment:
Do we not expect this problem to exist as well for future versions of windows server? I thought that it was a new issue introduced post-2019. If so I would think detecting whether there are multiple vs versions installed is a more robust method as opposed to just looking at OS versions.

Reply:
We expect this problem to be gone circa mid-May: https://github.com/actions/runner-images/issues/9701